File: HW2/householder.m
function H = householder(x)
    v = x(:);  

    % Calcolo di sigma
    sigma = sign(v(1)) * norm(v);

    % Calcolo di u (versore)
    u = v + sigma * eye(length(v), 1);
    u = u / norm(u);

    % Calcolo della matrice di riflessione
    H = eye(length(u)) - 2 * (u * u');

    return;
end

File: HW3/householder.py
import numpy as np

def householder(x):
    v = x.reshape(x.size, 1)

    sigma = np.sign(v[0, 0]) * np.linalg.norm(v)
    u = v + sigma * np.eye(v.size, 1)
    u = u / np.linalg.norm(u)
    
    Px = np.identity(u.size) - 2 * u @ u.T
    
    return Px